src/drivers/distance_sensor/broadcom/afbrs50/AFBRS50.cpp

```diff
@@ -101,6 +102,7 @@ void AFBRS50::ProcessMeasurement(void *data)
 				quality = 0;
 			}
 
+			_current_distance = result_m;
```

Comment:
Instead of storing the extra state you could do the mode update here. This likely also avoids any weird threading issues because the library backend is the one calling this.

Reply:
```suggestion
			UpdateMode(result_m);
```